src/components/Match/Match.js

```diff
@@ -78,20 +82,69 @@ const Match = ( {
             navToTaxonDetails={navToTaxonDetails}
           />
           { !suggestionsLoading
-    && (
-      <View className="mt-5">
-        <Heading3 className="mx-4">
-          {t( "Do-you-know-what-group-this-is-in" )}
-        </Heading3>
-        <IconicSuggestionsScroll
-          iconicTaxonChosen={iconicTaxon}
-          onIconicTaxonChosen={setIconicTaxon}
+          && (
+            <View className="mt-5">
+              <Heading3 className="mx-4">
+                {t( "Do-you-know-what-group-this-is-in" )}
+              </Heading3>
+              <IconicSuggestionsScroll
+                iconicTaxonChosen={iconicTaxon}
+                onIconicTaxonChosen={setIconicTaxon}
+              />
+              <Body2 className="mx-4 mt-7 mb-20">
+                {t( "If-you-took-the-original-photo-you-can-help" )}
+              </Body2>
+            </View>
+          )}
+        </ScrollViewWrapper>
+        <SaveDiscardButtons
+          handlePress={handleSaveOrDiscardPress}
         />
-        <Body2 className="mx-4 mt-7 mb-20">
-          {t( "If-you-took-the-original-photo-you-can-help" )}
-        </Body2>
-      </View>
-    )}
+      </>
+    );
+  }
+  // In case there are suggestions but no top suggestion
+  if ( !topSuggestion ) {
+    return (
+      <>
+        <ScrollViewWrapper scrollRef={scrollRef}>
+          <View className={cardClassTop}>
+            {
+              suggestionsLoading
+                ? (
+                  <ActivityIndicator size={33} />
+                )
+                : (
+                  <Body2>
+                    {t( "Were-not-confident-enough-to-make-an-ID" )}
+                  </Body2>
+                )
+            }
+          </View>
+          <PhotosSection
+            representativePhoto={topSuggestion?.taxon?.representative_photo}
+            taxon={taxon}
+            obsPhotos={obsPhotos}
+            navToTaxonDetails={navToTaxonDetails}
+          />
+          <View className="pt-2">
+            <AdditionalSuggestionsScroll
+              noTopSuggestion
+              onSuggestionChosen={onSuggestionChosen}
+              otherSuggestions={otherSuggestions}
+              suggestionsLoading={suggestionsLoading}
+            />
+            {!latitude && (
+              <Button
+                className="mx-5 mb-7"
+                level="neutral"
+                text={t( "ADD-LOCATION-FOR-BETTER-IDS" )}
```

Comment:
I tried tapping ADD LOCATION on the Ivy photo that was shared in the [dual-species issue](https://linear.app/inaturalist/issue/MOB-600/match-screen-special-handling-of-potential-dual-species-situations), but it didn't appear to do anything. Tapping the button shows me the same not confident screen and the ADD LOCATION button is still visible. It does pop up the location permission gate but I don't think it's fetching new suggestions after that.